src/utils/uid.util.js

```diff
@@ -1,47 +1,32 @@
-const hash = require('object-hash');
-/**
- * Return a random int, used by `utils.getUid()`.
- *
- * @param {Number} min
- * @param {Number} max
- * @return {Number}
- * @api private
- */
-function getRandomInt(min, max) {
-  return Math.floor(Math.random() * (max - min + 1)) + min;
-}
+const { randomBytes } = require('node:crypto');
 
 /**
- * Validates the date(s) and return the object containing the prefix and date.
- *
- * Prefix may contain 'ge, le, lt, gt'.
- *
- * @param {*} dates
+ * Generates a UUID that matches with FHIR UUID data type https://build.fhir.org/datatypes.html#uuid
```

Comment:
I noticed the regex comment you have and compared it to the original list of valid characters for this **uid** (not UUID).  Are the callers of getUid expecting a UUID?

I think the only thing needing to be updated is the old getRandomInt code.  Instead of using Math.random(), consider the example posted here:
https://stackoverflow.com/questions/18230217/javascript-generate-a-random-number-within-a-range-using-crypto-getrandomvalues